File: 백준/Silver/10773. 제로/제로.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int k = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();
        int answer = 0;
        while(k>0){
            int num = Integer.parseInt(br.readLine());
            if(num==0){
                if(!stack.isEmpty()){
                    answer-=stack.pop();  
                }
            }else{
                answer+=num;
                stack.push(num);
            }
            k--;
        }
        br.close();
        System.out.println(answer);
    }
}

File: 백준/Silver/10773. 제로/제로.py
import sys
# from collections import deque
input = sys.stdin.readline

k = int(input())
# dq = deque()
arr = []
answer = 0
for _ in range(k):
    num = int(input())
    if num == 0:
        answer -= arr.pop()
    else:
        answer += num
        arr.append(num)
print(answer)
